ui/components/component-library/avatar-badge/avatar-badge.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Box from '../../ui/box/box';
+
+export const badgePositions = ['top', 'bottom'];
+
+export const AvatarWithBadge = ({
+  children,
+  badgePosition,
+  badge,
+  ...props
+}) => {
+  return (
+    <Box className="avatar-badge" {...props}>
+      {/* Jazzicon , Avatar Account */}
+      {children}
+
+      <Box
+        className={
+          badgePosition === 'top'
+            ? 'avatar-badge-token-position-top'
+            : 'avatar-badge-token-position-bottom'
+        }
+      >
+        {badge}
+      </Box>
+    </Box>
+  );
+};
+
+AvatarWithBadge.propTypes = {
+  /**
+   * The position of the Badge
+   * Possible values could be 'top', 'bottom',
+   */
+  badgePosition: PropTypes.oneOf(badgePositions),
+  /**
+   * The children to be rendered inside the AvatarWithBadge
+   */
+  children: PropTypes.node,
+  /**
+   * The badge to be rendered inside the AvatarWithBadge
```

Comment:
Could add size suggestion here
```suggestion
   * The badge to be rendered inside the AvatarWithBadge
   * If using AvatarNetwork should be SIZES.XS
```